src/config.rs

```diff
@@ -59,37 +86,98 @@ impl fmt::Display for Device {
 }
 
 pub fn read_device(root: &Path, name: &str) -> Result<Option<Device>> {
-    match read_devices(root)?
-        .find(|(section_name, _)| section_name.as_ref().map(String::as_str) == Some(name))
-    {
-        Some((section_name, section)) => {
-            let memtotal_mb = get_total_memory_kb(root)? as f64 / 1024.;
-            parse_device(section_name, section, memtotal_mb)
-        }
-        None => Ok(None),
-    }
+    let memtotal_mb = (get_total_memory_kb(&root)? as f64 / 1024.) as u64;
+    return Ok(read_devices(root, memtotal_mb)?.remove(name));
 }
 
 pub fn read_all_devices(root: &Path) -> Result<Vec<Device>> {
-    let memtotal_mb = get_total_memory_kb(&root)? as f64 / 1024.;
-    Result::from_iter(
-        read_devices(root)?
-            .map(|(sn, s)| parse_device(sn, s, memtotal_mb))
-            .map(Result::transpose)
-            .flatten(),
-    )
+    let memtotal_mb = get_total_memory_kb(&root)? / 1024;
+
+    let devices: Vec<Device> = read_devices(root, memtotal_mb)?
+        .into_iter()
+        .filter(|(_, dev)| dev.disksize > 0)
```

Comment:
This is normally disallowed by zram.ko:
```
nabijaczleweli@tarta:~$ sudo tee /sys/block/zram0/disksize
0
0
tee: /sys/block/zram0/disksize: Invalid argument
```

So mentioning that this is an explicit config point "you can disable a device by setting its size to zero" or sth in the README would be nice.